railties/lib/rails/generators/named_base.rb

```diff
@@ -150,23 +150,23 @@ def redirect_resource_name # :doc:
         def model_resource_name(prefix: "") # :doc:
           resource_name = "#{prefix}#{singular_table_name}"
           if options[:model_name]
-            "[#{controller_class_path.map { |name| ":" + name }.join(", ")}, #{resource_name}]"
+            "[#{(controller_class_path.map { |name| ":" + name }.to_a << resource_name) * ', '}]"
```

Comment:
This will create an array with one element. But I think it is better to generate just elements instead of arrays. Just to check if `controller_class_path` is empty, how about?

```diff  
diff --git a/railties/lib/rails/generators/named_base.rb b/railties/lib/rails/generators/named_base.rb
index d6732f8..46c8f67 100644
--- a/railties/lib/rails/generators/named_base.rb
+++ b/railties/lib/rails/generators/named_base.rb
@@ -149,7 +149,7 @@ def redirect_resource_name # :doc:
 
         def model_resource_name(prefix: "") # :doc:
           resource_name = "#{prefix}#{singular_table_name}"
-          if options[:model_name]
+          if options[:model_name] && !controller_class_path.empty?
             "[#{controller_class_path.map { |name| ":" + name }.join(", ")}, #{resource_name}]"
           else
             resource_name
@@ -157,7 +157,7 @@ def model_resource_name(prefix: "") # :doc:
         end
 
         def singular_route_name # :doc:
-          if options[:model_name]
+          if options[:model_name] && !controller_class_path.empty?
             "#{controller_class_path.join('_')}_#{singular_table_name}"
           else
             singular_table_name
@@ -165,7 +165,7 @@ def singular_route_name # :doc:
         end
 
         def plural_route_name # :doc:
-          if options[:model_name]
+          if options[:model_name] && !controller_class_path.empty?
             "#{controller_class_path.join('_')}_#{plural_table_name}"
```

I think that it is better using the same checking in `singular_route_name` and `plural_route_name`.

Reply:
I completely agree, that would cleaner.

Reply:
@chiaraani Can you fix my pointed out?